### ISSUE TITLE
Revise compiler flags and the `--help` output for LLVM-by-default

### ISSUE THREAD

**Issue by e-kayrakli** [chapel-lang/chapel#18602]
Some of the compiler flags and the `--help` output intimate that we are still using C as our default backend generated code and the compiler. They need to be adjusted for LLVM as the default backend technology by either adding LLVM-specific flags and sections to the `--help` output, or making our flags backend-agnostic (or a mix of both). Things that stood out on a quick glance/discussion:

- `--ccflags`
- `--savec`
- Section names in the `--help` output
- `-g` sounds pretty specific about C code

**Comment by bradcray** [comment]
Ideally (and I'm not saying Engin disagrees, though it feels a little ambiguous in the OP), I think these flags and descriptions should not now be switched to reflect that LLVM is the default, but rather edited to be agnostic to the back-end / inclusive of both options).

**Comment by bradcray** [comment]
> I can also imagine that some flags are only applicable for the C backend and some only for the LLVM backend.

True.  Just wanted to make sure we didn't swing fully the other way into pretending that only the LLVM back-end exists.

**Comment by jabraham17** [comment]
Closing, this was improved by https://github.com/chapel-lang/chapel/pull/27918/

**Comment by bradcray** [comment]
It isn't clear to me that all the concerns Engin raised in the OP have been addressed.  Specifically, we still have `--savec` which implies generating C code which doesn't happen when using the LLVM back-end (yet the flag still has a purpose and role).  I think `--ccflags` still exists as well (though that one doesn't bother me as much personally).

**Comment by jabraham17** [comment]
`--savec` is explicitly documented as Saving generated Code, and mentions nothing about C. See https://chapel-lang.org/docs/main/usingchapel/man.html#index-70

And yes, `--ccflags` still exists, but its sibling `--mllvm` also exists

**Comment by bradcray** [comment]
I've always heard the LLVM-era concern about `--savec` being expressed as a concern about the name of the flag being misleading rather than its documentation being wrong.  So that's how I interpreted its being on this list—that the flag itself should be renamed.

[**edit:** or perhaps better for old-timers: given a synonym flag]

**Comment by bradcray** [comment]
I opened https://github.com/chapel-lang/chapel/issues/28796 to capture this last point.
